src/itemlist_formaction.cpp

```diff
@@ -109,6 +109,38 @@ void itemlist_formaction::process_operation(operation op, bool automatic, std::v
 		}
 	}
 	break;
+	case OP_OPENALLINBROWSER: {
+		int tabcount = 0;
+		LOG(LOG_INFO, "itemlist_formaction: opening all unread items in browser");
+		for (auto item : feed->items()) {
+			if (tabcount < v->get_cfg()->get_configvalue_as_int("max-browser-tabs")) {
+				if (item->unread()){
+					v->open_in_browser(item->link());
+					tabcount += 1;
+				}
+			} else {
+				break;
+			}
+		}
+	}
+	break;
+	case OP_OPENALLINBROWSER_AND_MARK:{
+		int tabcount = 0;
+		LOG(LOG_INFO, "itemlist_formaction: opening all unread items in browser and marking read");
+		for (auto item : feed->items()) {
+			if (tabcount < v->get_cfg()->get_configvalue_as_int("max-browser-tabs")) {
+				if (item->unread()){
+					v->open_in_browser(item->link());
+					item->set_unread(false);
+					tabcount += 1;
+				}
+			} else {
+				break;
+			}
+		}
```

Comment:
The code here is pretty much the same as in feedlist. Let's scrap the boilerplate, eh? (I realise the codebase doesn't yet look like we care, but we do. I'm gradually cleaning it up. You can help me by writing this one bit cleanly in the first place.)

You can create `include/formactions.h` and `src/formactions.cpp` and put this code there. Fit it into one function that accepts a feed and a boolean indicating if you should mark articles read. The logging code is better kept in `*_formaction.cpp` files as it's easier to grep the code then.